RecoTracker/ExoRegionalTracking/plugins/ExoRegionProducer.cc

```diff
@@ -0,0 +1,231 @@
+#include "RecoTracker/ExoRegionalTracking/plugins/ExoRegionProducer.h"
+
+ExoRegionProducer::ExoRegionProducer(const edm::ParameterSet &cfg, const ExoRegionTFCache *cache)
+    : rParam_(cfg.getParameter<double>("rParam")),
+
+      minRadius_(cfg.getParameter<double>("minRadius")),
+      discriminatorCut_(cfg.getParameter<double>("discriminatorCut")),
+      genMatch_(cfg.getParameter<string>("genMatch")),
+      input_names_(cfg.getParameter<vector<string> >("input_names")),
+      output_names_(cfg.getParameter<vector<string> >("output_names")) {
+  beamSpotToken_ = consumes<reco::BeamSpot>(cfg.getParameter<edm::InputTag>("beamSpot"));
+  trackClustersToken_ =
+      consumes<edm::View<reco::VertexCompositeCandidate> >(cfg.getParameter<edm::InputTag>("trackClusters"));
+  if (genMatch_ != "None")
+    genParticlesToken_ = consumes<edm::View<reco::GenParticle> >(cfg.getParameter<edm::InputTag>("genParticles"));
+
+  if (genMatch_ != "None" && genMatch_ != "HToSSTobbbb" && genMatch_ != "stopToBottom")
+    throw cms::Exception("BadConfig")
+        << "Parameter genMatch must be set to \"None\", \"HToSSTobbbb\", or \"stopToBottom\".";
+
+  unsigned nThreads = cfg.getParameter<unsigned>("nThreads");
+  string singleThreadPool = cfg.getParameter<string>("singleThreadPool");
+  tensorflow::SessionOptions sessionOptions;
+  tensorflow::setThreading(sessionOptions, nThreads, singleThreadPool);
+  session_ = tensorflow::createSession(cache->graphDef, sessionOptions);
+
+  produces<vector<reco::Vertex> >();
+}
+
+ExoRegionProducer::~ExoRegionProducer() {
+  if (session_ != nullptr)
+    tensorflow::closeSession(session_);
+}
+
+unique_ptr<ExoRegionTFCache> ExoRegionProducer::initializeGlobalCache(const edm::ParameterSet &cfg) {
+  tensorflow::setLogging("3");
+  string pbFile = cfg.getParameter<edm::FileInPath>("graph_path").fullPath();
+  ExoRegionTFCache *cache = new ExoRegionTFCache();
+  cache->graphDef = tensorflow::loadGraphDef(pbFile);
+
+  return unique_ptr<ExoRegionTFCache>(cache);
+}
+
+void ExoRegionProducer::globalEndJob(const ExoRegionTFCache *cache) {
+  if (cache->graphDef != nullptr)
+    delete cache->graphDef;
+}
+
+void ExoRegionProducer::produce(edm::Event &event, const edm::EventSetup &setup) {
+  edm::Handle<reco::BeamSpot> beamSpot;
+  edm::Handle<edm::View<reco::VertexCompositeCandidate> > trackClusters;
+  edm::Handle<edm::View<reco::GenParticle> > genParticles;
+
+  event.getByToken(beamSpotToken_, beamSpot);
+  const math::XYZVector bs(beamSpot->x0(), beamSpot->y0(), beamSpot->z0());
+  event.getByToken(trackClustersToken_, trackClusters);
+  vector<math::XYZVector> decayVertices;
+  if (genMatch_ != "None") {
```

Comment:
@slava77 @jpata is it ok to have in RECO code some dependency on gen ?

Reply:
IIRC, we do have some examples of dependence on MC truth objects in reco modules. 
Still, it is not really a welcome dependence and if possible, it would be better to decouple.
clearly, the default configuration is not supposed to use MC-specific inputs.

Reply:
It makes sense to decouple the MC truth dependence, which was only used for testing anyway, so I've removed everything related to `genMatch_`.